console/console-init/ui/src/Components/AddressSpace/AddressListFilter.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from "react";
+import {
+  DropdownComponent,
+  IDropdownOption
+} from "../Common/DropdownComponent";
+import {
+  InputGroup,
+  Button,
+  TextInput,
+  ButtonVariant
+} from "@patternfly/react-core";
+import { SearchIcon } from "@patternfly/react-icons";
+
+export interface IAddressListFilterProps {
+  onSearch: () => void;
+  isFilterDropdownOpen: boolean;
+  onFilterSelect: (item: any) => void;
+  filterValue: string;
+  setFilterOpen: () => void;
+  isTypeDropdownOpen: boolean;
+  onTypeSelect: (item: any) => void;
+  typeValue: string;
+  setTypeOpen: () => void;
+  isStatusDropdownOpen: boolean;
+  onStatusSelect: (item: any) => void;
+  statusValue: string;
+  setStatusOpen: () => void;
+}
+export const filterOptions: IDropdownOption[] = [
+    { value: "name", label: "Name" },
+    { value: "type", label: "Type" },
+    { value: "status", label: "Status" }
+  ],
+  typeOptions: IDropdownOption[] = [
+    { value: "queue", label: "Queue" },
+    { value: "topic", label: "Topic" },
+    { value: "subscripition", label: "Subscripition" },
+    { value: "mulitcast", label: "Mulitcast" },
+    { value: "anycast", label: "Anycast" }
+  ],
+  statusOptions: IDropdownOption[] = [
+    { value: "active", label: "Active" },
+    { value: "configuring", label: "Configuring" },
+    { value: "failed", label: "Failed" }
+  ];
+
+export const AddressListFilter: React.FunctionComponent<
+  IAddressListFilterProps
+> = ({
+  onSearch,
+  isFilterDropdownOpen,
+  onFilterSelect,
+  filterValue,
+  setFilterOpen,
+  isTypeDropdownOpen,
+  onTypeSelect,
+  typeValue,
+  setTypeOpen,
+  isStatusDropdownOpen,
+  onStatusSelect,
+  statusValue,
+  setStatusOpen
+}) => {
+  const DataToRender = () => {
```

Comment:
What's the value of this being a function?

Can't the returned JSX elements be components instead with their own props? The DataToRender function would then be the logic to render the right component, which could be in the render function itself:
```tsx
{filterValue === "Name" && <NameFilter foo={foo} />}
{filterValue === "Type" && <TypeFilter bar={bar} />}
etc
```